paddle/go/pserver/service.go

```diff
@@ -135,7 +138,10 @@ func (s *Service) SendGrads(grads []Gradient, dummy *int) error {
 	errCh := make(chan error, count)
 	for _, g := range grads {
 		go func(p Parameter, g Gradient) {
-			err := s.opt.UpdateParameter(p, g)
+			opt, err := s.optimizerMap[p.Name]
```

Comment:
I noticed the code here does not compile, because `err` is `bool` type, but you can not check `bool` type against `nil` (next line).

We should not check in code that does not compile. Maybe I can do the Go code part for now, and you can get more familiar with Go by reviewing the PR?